libs/pika/command_line_handling/src/command_line_handling.cpp

```diff
@@ -967,6 +961,37 @@ namespace pika::detail {
             return 1;
         }
 
+        // print a warning about process masks resulting in only one worker
+        // thread
+        if (!ignore_process_mask_)
+        {
+            if (num_threads_ == 1)
+            {
+                std::cerr
+                    << "The pika runtime will be started with only one worker "
+                       "thread because the process mask has restricted the "
+                       "available resources to only one thread or the system "
+                       "only has one core. If this is unintentional make sure "
+                       "the process mask contains the resources you need or "
+                       "use --pika:ignore-process-mask to use all resources. "
+                       "Use --pika:print-bind to print the thread bindings "
+                       "used by pika.\n";
+            }
```

Comment:
Will this warning be printed even when the user specifies `pika:threads=1`? I would find this annoying.

Maybe it also worth to print the message only if the system has more than one core.

Reply:
Yes, it would currently do that. These are good suggestions and I'll update this to take that into account.

Reply:
@rasolca I've udpated the conditions for printing the warning message. It is now printed only if:
1. The number of worker threads (or cores) is 1,
2. The default number of worker threads is bigger than one (i.e. what you would get without a process mask), and
3. `--pika:threads` (or `--pika:cores`) is not given as a command line argument.

I think that should make it print only in cases where one might be surprised that the number of worker threads is 1.

Reply:
Thanks!